free/src/test/scala/cats/free/FreeApplicativeTests.scala

```diff
@@ -56,6 +64,17 @@ class FreeApplicativeTests extends CatsSuite {
     r1.foldMap(nt) should === (r2.foldMap(nt))
   }
 
+  test("FreeApplicative#flatCompile") {
```

Comment:
Doing this has exposed a `ClassCastException` bug in the implementation, which I'm working through now.